### DISCUSSION TITLE
Question about regular expression in V and string(numbers)

### DISCUSSION THREAD

**Discussion by GGRei** [vlang/v#14315] [General]
Hello,

I try to define with a simple regex operation if some inputs numbers have a valid format. ( integer/decimal). The specificity of my test  should use "period" character and not "comma" for the decimal part.

I work with this regular expression and the **matches_string(string)bool** function ( native regex module function ) :

`([0-9]+[.]{1}[0-9]+)|([0-9]+[^.])`

However, I have problems in some situations like  :

`2000 is invalid (first problem), 15.50 is valid( OK here ), .1500 is invalid (OK here ), 2000. is invalid ( OK here ), 20..00 is invalid ( OK here ),  20.3.50 is valid (second problem here ), 20.0.0. is valid ( third problem here ), 200da0 or 20 36 is valid ( fourth problem here ). 
etc.....`

I went through the documentation : [V regex module](https://modules.vlang.io/regex.html), and use some stuff like print of the regex execution code, but I admit to being in trouble with my test.

In javascript, with this regex : 

`^[0-9]+\.?[0-9]+$/gm`

I get better results : [test javascrip url](https://www.regextester.com/?fam=122400)

If someone have an idea and can explain at how resolve this issue with V regex. Thanks in advance!

**Comment by JalonSolov** [comment]
Can you show some example code?

**Comment by GGRei** [comment]
Sure :

```
import regex

fn test_regex(){

	query:= '([0-9]+[.]{1}[0-9]+)|([0-9]+[^.])'  //you can use my javascript example : ^[0-9]+\.?[0-9]+$
	mut re := regex.regex_opt(query) or { panic(err) }

	println(re.get_code())
        
        // can use : 2000 or 2000. or .2000 or 20.20.0 or 15ds300 etc....
	if re.matches_string("2000") {
		println("valid integer")
	}else{
		println("invalid integer")
	}

	if re.matches_string("15.00") {
		println("valid decimal")
	}else{
		println("invalid decimal")
	}

}

test_regex()
```

**Comment by GGRei** [comment]
Well, after reading again the V doc and some test, I finded the good V regex : 

`^(([0-9])+([.]){1}([0-9])+)|(([0-9])+)$`
